feat: 디테일 페이지 구현 (#98)

tkdgheee opened the pull request:
## 🔥 Description (필수)

이 PR이 어떤 작업을 했는지 명확하게 작성해주세요.  
**최소 3줄 이상** 상세하게 작성 부탁드립니다.

- Detail 컴포넌트를 8개의 작은 컴포넌트로 분리 (단일 책임 원칙 적용)
- 비즈니스 로직을 useDetailPage 커스텀 훅으로 분리
- 하드코딩된 상수들을 detailOption로 중앙화
- 동적 날짜 선택 기능 추가 (14일 범위)
- feedback mock 데이터 수정
---

## ⏰ Due (필수)

- 리뷰 마감 요청일: **2025.08.14까지 리뷰 부탁드립니다.**

---

## 🔄 변경 사항 (필수)

- [ ] 기능 추가 (Feature)
- [ ] 버그 수정 (Bug Fix)
- [ ] 리팩토링 (Refactor) — 기능 변경 없이 코드 개선
- [ ] UI/스타일 수정 (UI/Style) — 기능 영향 없는 스타일, 디자인 변경
- [ ] 문서 수정 (Docs) — README, 문서화 등
- [ ] 테스트 코드 추가/수정 (Test)
- [ ] 배포 설정/환경 설정 변경 (Chore/Infra)
- [ ] 성능 최적화 (Performance)
- [ ] 기타 (Other)

---

## ✔️ 변경 사항 상세 설명

- ProductImage: 상품 이미지 표시
- PriceInfo: 가격 및 리뷰 정보 표시  
- QuantitySelector: 수량 선택 기능
- OptionsSection: 픽업 옵션 선택
- MenuTab: 상세정보 탭 메뉴
- ProductInfoTable: 상품 정보 테이블
- ReviewSection: 상품 후기 표시

Hooks:
- useDetailPage: 상태 관리 및 비즈니스 로직 처리

tkdgheee committed:
feat: 디테일 페이지 구현 (#97)

- Detail 컴포넌트를 8개의 작은 컴포넌트로 분리 (단일 책임 원칙 적용)
- 비즈니스 로직을 useDetailPage 커스텀 훅으로 분리
- 하드코딩된 상수들을 detailOptions로 중앙화
- 동적 날짜 선택 기능 추가 (14일 범위)
- feedback mock 핸들러와 mock 데이터 api 명세서 양식에 맞게 수정

tkdgheee committed:
feat: 디테일 페이지 구현 (#98)

- 리베이스 충돌 사항 수정

tkdgheee committed:
feat: 디테일 페이지 구현 (#98)

- 리베이스 충돌 사항 수정

tkdgheee committed:
feat: 디테일 페이지 구현 (#98)

- 리베이스 충돌 사항 수정
- 변경된 피드백 목록 조회 파라미터명에 따른 파일 수정

devjaesung commented on src/mocks/handlers/feedback/mocks/feedbackListMockData.ts:
```diff
@@ -1,106 +1,74 @@
-const feedbackListMockData = {
-  count: 15,
-  next: 'http://api.example.com/api/v1/feedbacks/?page=2',
-  previous: null,
-  results: [
-    {
-      id: 123,
```
소연님 파트에서 사용되는 목데이터인데 수정하신 이유가 있으실까요?

devjaesung reviewed (changes requested):
지금 mockdata에 있는 id로 페이지 이동해도 msw가 데이터를 찾지 못해서 페이지가 안 보입니다! msw부분 확인 한 번 해주시고 상품 카드 클릭하면 페이지 이동될 수 있게 링크도 추가해주세요!

tkdgheee committed:
feat: 패키지 페이지 구현 (#98)

- CardBase 컴포넌트에 pacakge, product 링크 추가
- cardProps, pacakge.ts productUtils에 type 추가

tkdgheee committed:
feat: 디테일 페이지 구현 (#98)

- 목 데이터와 타입 파일 id값을 string으로 변경

tkdgheee committed:
feat: 디테일 페이지 구현 (#98)

- 피드백 mock 데이터 스웨거에 맞게 수정
- feedback 핸들러에 있던 타입을 feedback.ts 파일로 이동

tkdgheee committed:
feat: 디테일 페이지 구현 (#98)

- import 경로 수정
